include/cocaine/context.hpp

```diff
@@ -152,6 +169,9 @@ class context_t {
     // because services are allowed to start and stop other services during their lifetime.
     synchronized<service_list_t> m_services;
 
+    //Flag set after initialization
+    bool ready;
```

Comment:
In my opinion, boolean flags are usually a manifestation of a bad design. I'd rather accept somewhat larger refactoring, but not a bool flag crutch.